src/base/static/components/organisms/tag-bar.js

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "react-emotion";
+import { connect } from "react-redux";
+
+import Tag from "../molecules/tag";
+import TagEditor from "../molecules/tag-editor";
+
+import {
+  getAllTagsForDataset,
+  getTagDisplayName,
+  getColorForTag,
+  getTagFromUrl,
+} from "../../state/ducks/datasets";
+
+const TagBarContainer = styled("div")({
+  borderBottom: "2px solid #ddd",
+  marginBottom: "16px",
+  paddingBottom: "8px",
+});
+
+const TagBar = props => {
+  return props.isEditModeToggled ? (
+    <TagBarContainer>
+      {props.getAllTagsForDataset(props.datasetSlug).map(tag => {
+        return (
+          <TagEditor
+            key={tag.id}
+            datasetSlug={props.datasetSlug}
+            onUpdateTagNote={props.onUpdateTagNote}
+            onDeletePlaceTag={props.onDeletePlaceTag}
+            onCreatePlaceTag={props.onCreatePlaceTag}
+            backgroundColor={tag.color}
+            displayName={props.getTagDisplayName(props.datasetSlug, tag.id)}
+            tag={tag}
+            placeTag={props.placeTags.find(
+              placeTag => placeTag.tag === tag.url,
+            )}
+            placeUrl={props.placeUrl}
+          />
+        );
+      })}
+    </TagBarContainer>
+  ) : (
+    props.placeTags.length > 0 && (
+      <TagBarContainer>
+        {props.placeTags.map(placeTag => {
+          return (
+            <Tag
+              key={placeTag.id}
+              datasetSlug={props.datasetSlug}
+              backgroundColor={props.getColorForTag(
+                props.datasetSlug,
+                placeTag.tag,
+              )}
+              placeTag={placeTag}
+              displayName={props.getTagDisplayName(
+                props.datasetSlug,
+                props.getTagFromUrl(props.datasetSlug, placeTag.tag).id,
+              )}
+            />
+          );
+        })}
+      </TagBarContainer>
+    )
+  );
+};
+
+TagBar.propTypes = {
+  datasetSlug: PropTypes.string.isRequired,
+  getAllTagsForDataset: PropTypes.func.isRequired,
+  getTagDisplayName: PropTypes.func.isRequired,
+  getColorForTag: PropTypes.func.isRequired,
+  getTagFromUrl: PropTypes.func.isRequired,
+  isEditModeToggled: PropTypes.bool.isRequired,
+  onDeletePlaceTag: PropTypes.func.isRequired,
+  onCreatePlaceTag: PropTypes.func.isRequired,
+  onUpdateTagNote: PropTypes.func.isRequired,
+  placeTags: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      comment: PropTypes.string,
+    }),
+  ),
```

Comment:
Since we are using this prop type in multiple parts of the code, I suggest saving this into a `placeTagsPropType` variable in our Places duck, and importing it here.